src/models/classes/container.js

```diff
@@ -36,16 +36,16 @@ export default class Container extends StorageCore
      *
      * @override
      */
-    removeItem(product, perishedQuantity)
+    removeItem(product, perishedProduct)
     {
         this.items = this.items.filter(function (item) {
             if (product.name != item.name) {
                 return item;
             }
 
-            if (product.values.isPerishable === true) {
-                const removedQuantity = Math.min(product.values.quantity, perishedQuantity)
-             
+            if (product.values.isPerishable) {
+                const removedQuantity = perishedProduct.values.quantity;
```

Comment:
`Math.min(product.values.quantity, item.values.quantity, perishedQuantity);`

You can only delete the bare minimum of this (and you should not use `perishQuantity`, but rather update the value on `product` outside this function.

Reply:
Yeah that was dumb of me. I now made it 

```Math.min(product.values.quantity, perishedProduct.values.quantity);```

But I just keep struggling with that it doesn't stop removing from a container after the first object has been removed (when I want it to).